common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -93,7 +93,7 @@
  * This {@code classpath} configuration is only supported on YARN versions >= 2.9.0.
  */
 public class YarnShuffleService extends AuxiliaryService {
-  private static final Logger logger = LoggerFactory.getLogger(YarnShuffleService.class);
+  private static Logger logger = LoggerFactory.getLogger(YarnShuffleService.class);
```

Comment:
To clarify, this should be `final` - the `defaultLogger` from earlier PR.
Add a new `private Logger logger` - which is initialized to `defaultLogger` by default; and overridden in `serviceInit` as required.